vendor/github.com/lafriks/xormstore/xormstore.go

```diff
@@ -0,0 +1,245 @@
+/*
+Package xormstore is a XORM backend for gorilla sessions
+
+Simplest form:
+
+	store := xormstore.New(engine, []byte("secret-hash-key"))
+
+All options:
+
+	store := xormstore.NewOptions(
+		engine, // *xorm.Engine
+		xormstore.Options{
+			TableName: "sessions",  // "sessions" is default
+			SkipCreateTable: false, // false is default
+		},
+		[]byte("secret-hash-key"),      // 32 or 64 bytes recommended, required
+		[]byte("secret-encyption-key")) // nil, 16, 24 or 32 bytes, optional
+
+		// some more settings, see sessions.Options
+		store.SessionOpts.Secure = true
+		store.SessionOpts.HttpOnly = true
+		store.SessionOpts.MaxAge = 60 * 60 * 24 * 60
+
+If you want periodic cleanup of expired sessions:
+
+		quit := make(chan struct{})
+		go store.PeriodicCleanup(1*time.Hour, quit)
+
+For more information about the keys see https://github.com/gorilla/securecookie
+
+For API to use in HTTP handlers see https://github.com/gorilla/sessions
+*/
+package xormstore
+
+import (
+	"encoding/base32"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/lafriks/xormstore/util"
+
+	"github.com/go-xorm/xorm"
+	"github.com/gorilla/context"
+	"github.com/gorilla/securecookie"
+	"github.com/gorilla/sessions"
+)
+
+const sessionIDLen = 32
+const defaultTableName = "sessions"
+const defaultMaxAge = 60 * 60 * 24 * 30 // 30 days
+const defaultPath = "/"
+
+// Options for xormstore
+type Options struct {
+	TableName       string
+	SkipCreateTable bool
+}
+
+// Store represent a xormstore
+type Store struct {
+	e           *xorm.Engine
+	opts        Options
+	Codecs      []securecookie.Codec
+	SessionOpts *sessions.Options
+}
+
+type xormSession struct {
+	ID          string         `xorm:"VARCHAR(400) PK NAME 'id'"`
```

Comment:
`VARCHAR(400)` results in an error when I use mysql:
```
panic: Error 1071: Specified key was too long; max key length is 767 bytes

goroutine 1 [running]:
code.gitea.io/gitea/vendor/github.com/lafriks/xormstore.NewOptions(0xc4206680b0, 0x113ce57, 0xf, 0xc42067d600, 0xc4206fb8f0, 0x1, 0x1, 0x0)
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/vendor/github.com/lafriks/xormstore/xormstore.go:108 +0x2d2
code.gitea.io/gitea/modules/auth/oauth2.Init(0xc4206680b0)
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/modules/auth/oauth2/oauth2.go:44 +0xb9
code.gitea.io/gitea/models.InitOAuth2()
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/models/oauth2.go:101 +0x3f
code.gitea.io/gitea/routers.GlobalInit()
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/routers/init.go:63 +0x56e
code.gitea.io/gitea/cmd.runWeb(0xc420344000, 0x0, 0x0)
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/cmd/web.go:83 +0xa8
code.gitea.io/gitea/vendor/github.com/urfave/cli.HandleAction(0xfb0900, 0x117e690, 0xc420344000, 0xc42020f440, 0x0)
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:471 +0xad
code.gitea.io/gitea/vendor/github.com/urfave/cli.(*App).Run(0xc4200d5040, 0xc420030190, 0x1, 0x1, 0x0, 0x0)
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:246 +0x542
main.main()
	/mnt/local_storage/jonasfranz/Go/src/code.gitea.io/gitea/main.go:52 +0x40e
```
The error is returned at line 106 and is actually not handled!

When I change this to `VARCHAR(200)` for example, no error is thrown.